From 11b975718f4e1a9ceef9b11be252e479d54e8f0f Mon Sep 17 00:00:00 2001
From: Zen <z@pyl.onl>
Date: Tue, 4 Feb 2025 17:03:46 -0600
Subject: [PATCH 01/58] kmod: mask i915 by default

Signed-off-by: Zen <z@pyl.onl>
---
 src/ugrd/kmod/novideo.toml | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/ugrd/kmod/novideo.toml b/src/ugrd/kmod/novideo.toml
index c556c5f..2a20897 100644
--- a/src/ugrd/kmod/novideo.toml
+++ b/src/ugrd/kmod/novideo.toml
@@ -5,5 +5,6 @@ kmod_ignore = [
 'amdgpu', 'radeon', 'amdxcp',  # AMD GPU
 'nouveau', 'nvidia', 'nvidia_drm',  # NVIDIA GPU
 'ast', 'mgag200',  # Server GPU
+'i915',  # Intel GPU
 'videobuf2_common', 'videobuf2_v4l2', 'drm_suballoc_helper', 'videobuf2_vmalloc', 'videobuf2_memops',  # V4L2
 ]
-- 
2.48.1

