From 5802f46b0e61f5516fffd554d0ee71090d7d5abe Mon Sep 17 00:00:00 2001
From: Zen <z@pyl.onl>
Date: Mon, 17 Feb 2025 12:47:21 -0600
Subject: [PATCH 54/58] remove unused late_fstab

Signed-off-by: Zen <z@pyl.onl>
---
 src/ugrd/fs/mounts.toml | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/ugrd/fs/mounts.toml b/src/ugrd/fs/mounts.toml
index 68b9ec3..c1affae 100644
--- a/src/ugrd/fs/mounts.toml
+++ b/src/ugrd/fs/mounts.toml
@@ -21,7 +21,6 @@ autodetect_root_luks = true
 autodetect_root_lvm = true
 autodetect_root_raid = true
 autodetect_init_mount = true
-late_fstab = "/etc/fstab.late"
 
 [imports.config_processing]
 "ugrd.fs.mounts" = [ "_process_run_dirs_multi",
@@ -65,7 +64,6 @@ mounts = "dict"  # Add the mounts property, used to define the mounts to be made
 mount_devpts = "bool"  # Whether or not to mount devpts
 run_dirs = "NoDupFlatList"  # A list of directories to be created under /run
 late_mounts = "dict"  # Like mounts, but run after the root is mounted
-late_fstab = "str"  # The path to the late_fstab file
 auto_mounts = "NoDupFlatList"  # A list of mounts to be automatically added to the mounts list
 mount_timeout = "float"  # The time to wait between mount attempts
 mount_retries = "int"  # The number of times to re-attempt mounting the fstab, infinite if not set
-- 
2.48.1

