From 8859e51cc18d4ad7f4512737801f5ea07639a660 Mon Sep 17 00:00:00 2001
From: Zen <z@pyl.onl>
Date: Mon, 17 Feb 2025 16:59:52 -0600
Subject: [PATCH 57/58] fix accidental capitalization

Signed-off-by: Zen <z@pyl.onl>
---
 src/ugrd/base/base.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ugrd/base/base.py b/src/ugrd/base/base.py
index c4d51ea..389e750 100644
--- a/src/ugrd/base/base.py
+++ b/src/ugrd/base/base.py
@@ -127,7 +127,7 @@ def do_switch_root(self) -> str:
         _find_init || rd_fail "Unable to find init."  # Redefines init on success
         init_target=$(readvar init)
     fi
-    einfo "Switching root to: $switch_root_Target $init_target"
+    einfo "Switching root to: $switch_root_target $init_target"
     klog "[UGRD $(readvar VERSION)] Running init: $init_target"
     einfo "Cleaning up /run/ugrd"
     edebug "$(rm -rfv /run/ugrd)"
-- 
2.48.1

